pkg/network/protocols/amqp/client.go

```diff
@@ -173,3 +176,50 @@ func (c *Client) Consume(queue string, numberOfMessages int) ([]string, error) {
 
 	return res, nil
 }
+
+// newAMQPConnection wraps connection creation from the "amqp" package. It handles
+// the differences in the connection creation process between plaintext & TLS
+// connections. Specifically, for TLS connections, it uses a
+// TransparentUnixProxyServer to handle the TLS part, allowing tests to hook it
+// using USM's GoTLS decoding.
+// Returns a new connection to the AMQP server, on an error if it failed to
+// make one.
+func newAMQPConnection(opts Options) (*amqp.Connection, error) {
+	url := getURL(opts)
```

Comment:
url is relevant only when TLS is not enabled
thus, please move the declaration near the usage, and I think you should modify `getURL` accordingly, to support only plaintext (and maybe, remove the function, as it wraps a single `fmt.Sprintf` call)